titiler/pgstac/dependencies.py

```diff
@@ -26,13 +27,76 @@
 def SearchIdParams(
     search_id: Annotated[
         str,
-        Path(description="Search Id (pgSTAC Search Hash)"),
+        Path(description="PgSTAC Search Identifier (Hash)"),
     ]
 ) -> str:
     """search_id"""
     return search_id
 
 
+@cached(  # type: ignore
+    TTLCache(maxsize=cache_config.maxsize, ttl=cache_config.ttl),
+    key=lambda pool, collection_id: hashkey(collection_id),
+)
+@retry(
+    tries=retry_config.retry,
+    delay=retry_config.delay,
+    exceptions=(
+        pgErrors.OperationalError,
+        pgErrors.InterfaceError,
+    ),
+)
+def get_collection_id(pool: ConnectionPool, collection_id: str) -> str:
+    """Get Search Id for a Collection."""
+    search = model.PgSTACSearch(
+        filter={
+            "op": "=",
+            "args": [{"property": "collection"}, collection_id],
+        },
+        filter_lang="cql2-json",  # type: ignore
+    )
+
+    with pool.connection() as conn:
+        with conn.cursor(row_factory=dict_row) as cursor:
+            cursor.execute(
+                "SELECT * FROM pgstac.get_collection(%s);",
+                (collection_id,),
+            )
+            collection = cursor.fetchone()["get_collection"]
+            if not collection:
+                raise MosaicNotFoundError(f"CollectionId `{collection_id}` not found")
+
+            bbox = collection["extent"]["spatial"].get("bbox", [[-180, -90, 180, 90]])
+            metadata = model.Metadata(
+                name=f"Mosaic for '{collection_id}' Collection",
+                bounds=bbox[0],
+                # TODO: use asset extension to populate the `assets` attribute
+            )
+
+            cursor.row_factory = class_row(model.Search)
+            cursor.execute(
+                "SELECT * FROM search_query(%s, _metadata => %s);",
+                (
+                    search.model_dump_json(by_alias=True, exclude_none=True),
+                    metadata.model_dump_json(exclude_none=True),
+                ),
+            )
+            search_info = cursor.fetchone()
```

Comment:
Here we do two things:
- first check if the `collection_id` is valid (using `pgstac.get_collection` method). This will also be useful to retrieve collection's metadata (like done with the bbox). 

- register a search query to pgstac and return the search_id

cc @bitner